azure/iac/disks.rego

```diff
@@ -79,3 +79,102 @@ disk_encrypt_metadata := {
     "Policy Help URL": "",
     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/disks"
 }
+
+# PR-AZR-ARM-DSK-002
+#
+
+default disk_encryption_2 = null
+
+
+azure_attribute_absence["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    not resource.properties.encryption.diskEncryptionSetId
+}
+
+source_path[{"disk_encryption_2":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.compute/disks"
+    not resource.properties.encryption.diskEncryptionSetId
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","diskEncryptionSetId"]]
+    }
+}
+
+azure_attribute_absence["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    not resource.properties.encryption.type
+}
+
+source_path[{"disk_encryption_2":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.compute/disks"
+    not resource.properties.encryption.type
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","diskEncryptionSetId"]]
+    }
+}
+
+azure_issue["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    count(resource.properties.encryption.diskEncryptionSetId) == 0
+}
+
+source_path[{"disk_encryption_2":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.compute/disks"
+    count(resource.properties.encryption.diskEncryptionSetId) == 0
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","diskEncryptionSetId"]]
+    }
+}
+
+azure_issue["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    lower(resource.properties.encryption.type) != "encryptionatrestwithcustomerkey"
```

Comment:
@shahinshirvani EncryptionAtRestWithCustomerKey and EncryptionAtRestWithPlatformAndCustomerKeys both are considerate as encryption with custom key. so in your case there will be an azure issue if only the encryption type is EncryptionAtRestWithPlatformKey
See the available types at https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/disks?tabs=bicep#encryption

The document you shared does not tell any different.
image
Goto the azure portal and you will see all the options there.